invenio_rdm_records/oaiserver/services/services.py

```diff
@@ -0,0 +1,226 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 Graz University of Technology.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""OAI-PMH service."""
+
+from flask import current_app
+from flask_babelex import lazy_gettext as _
+from flask_sqlalchemy import Pagination
+from invenio_oaiserver.models import OAISet
+from invenio_records_resources.services import Service
+from invenio_records_resources.services.base import LinksTemplate
+from invenio_records_resources.services.records.schema import \
+    ServiceSchemaWrapper
+from invenio_records_resources.services.uow import unit_of_work
+from sqlalchemy.orm.exc import NoResultFound
+from sqlalchemy.sql import text
+
+from invenio_rdm_records.oaiserver.services.errors import \
+    OAIPMHSetDoesNotExistError, OAIPMHSetIDDoesNotExistError, \
+    OAIPMHSetSpecAlreadyExistsError
+from invenio_rdm_records.oaiserver.services.uow import OAISetCommitOp, \
+    OAISetDeleteOp
+
+
+class OAIPMHServerService(Service):
+    """OAI-PMH service."""
+
+    def __init__(self, config):
+        """Init service with config."""
+        super().__init__(config)
+
+    @property
+    def schema(self):
+        """Returns the data schema instance."""
+        return ServiceSchemaWrapper(self, schema=self.config.schema)
+
+    @property
+    def update_schema(self):
```

Comment:
Due to https://github.com/inveniosoftware/invenio-oaiserver/blob/725f80056f05fefbfcc90e0994694230f38fbe9d/invenio_oaiserver/models.py#L68-L73, it is not possible to update the spec which is why it is not included in the update schema. It would throw an error eventually so it would be possible to reduce it to one schema.

Reply:
updated to only have one schema